{{cookiecutter.project_name}}/{% if cookiecutter.backend=='mesonpy' %}meson.build{% endif %}

```diff
@@ -1,34 +1,28 @@
 project(
-  '{{ cookiecutter.__project_slug }}',
-  'cpp',
-  version: '0.1.0',
-  license: '{{ cookiecutter.license }}',
-  meson_version: '>= 0.64.0',
-  default_options: [
-    'buildtype=debugoptimized',
-    'cpp_std=c++11',
-  ],
+    '{{ cookiecutter.__project_slug }}',
+    'cpp',
+    version: '0.1.0',
+    license: 'BSD',
+    meson_version: '>= 1.1.0',
+    default_options: [
+        'cpp_std=c++11',
+    ],
 )
-name = '{{ cookiecutter.__project_slug }}'
 
-py_mod = import('python')
-py = py_mod.find_installation(pure: false)
-
-pybind11_config = find_program('pybind11-config')
-pybind11_config_ret = run_command(pybind11_config, ['--includes'], check: true)
-pybind11 = declare_dependency(
-    include_directories: [pybind11_config_ret.stdout().split('-I')[-1].strip()],
-)
-
-install_subdir('src' / name, install_dir: py.get_install_dir() / name, strip_directory: true)
+py = import('python').find_installation(pure: false)
+pybind11_dep = dependency('pybind11')
 
 py.extension_module('_core',
     'src/main.cpp',
     subdir: '{{ cookiecutter.__project_slug }}',
     install: true,
-    dependencies : [pybind11],
-    link_language : 'cpp',
-    override_options: [
-        'cpp_rtti=true',
-    ]
+    dependencies : [pybind11_dep],
+)
+
+py.install_sources(
+    [
+        'src/{{ cookiecutter.__project_slug }}/__init__.py',
+        'src/{{ cookiecutter.__project_slug }}/_main.py',
```

Comment:
Note that since I didn't find the source files, I completely made up `_main.py` - it likely doesn't exist.

Reply:
I'm not sure how that works exactly, but in meson-python it would be a little tricky to get that right. Installing all `.py` files is wrong, and the `.gitattributes` method doesn't quite work either because that would also exclude `.py` files from the sdist.

Reply:
We rip the procedure straight from hatchling. It's based on a "package" list. The default is `["<package_name>", "src/<package_name>", "python/<package_name>"]` (last of which is unique to scikit-build-core). A user can specify their own custom list, including an empty list, which disables the process. Anything in these packages is included entirely (there is a `wheel.exclude` option too for filtering). Anything more exotic than packages needs to be copied around via CMake and won't work as well with editable installs (hatchling has a powerful force-include dict that we might also add a some point).

Reply:
Thanks for the context! I'll have to look into that, it does sound potentially useful.